libflux/src/ast/check/tests.rs

```diff
@@ -0,0 +1,114 @@
+use super::*;
+use crate::parser::parse_string;
+use crate::ast::{Position, File, BaseNode, VariableAssgn, Identifier, IntegerLit};
+use crate::ast::Statement::Variable;
+use crate::ast::Expression::Integer;
+
+#[test]
+fn test_object_check() {
+    let file = parse_string("object_test", "a = 1\nb = {c: 2, a}");
+    let got = check(walk::Node::File(&file));
+    let want = vec![Error {
+        location: SourceLocation {
+            file: Some(String::from("object_test")),
+            start: Position {
+                line: 2,
+                column: 5
+            },
+            end: Position {
+                line: 2,
+                column: 14
+            },
+            source: Some(String::from("{c: 2, a}"))
+        },
+        message: String::from("cannot mix implicit and explicit properties")
+    }];
+    assert!(want.iter().zip(got).all(|(want, got)| want == &got));
```

Comment:
if you change this to `assert_eq!(want, got)` does it work? If so I would prefer that method as Rust assert_eq method has good diffing logic for what is different between the values.

Reply:
Done